### DISCUSSION TITLE
Tools : choose to ping IPv4 or IPv6

### DISCUSSION THREAD

**Discussion by macjl** [isontheline/pro.webssh.net#1199] [Ideas]
Hello,

With the ping tool, I didn’t found a way to choose to ping a hostname with IPv4 or IPv6 protocol. It always seems to choose the IPv4, if the A record exists.

I also searched in mashREPL, and there is no ‘ping6’ command.

Could you add a way to force the protocol to use ?

**Comment by isontheline** [comment]
Bonjour Jean-Laurent 👋

No problem I will add the ability to choose between IPv4 and IPv6 DNS records in the user interface.
Regarding mashREPL I will try to import ping6 command.

I will start by the UI and will come back to you as soon as possible : https://github.com/isontheline/pro.webssh.net/issues/1200

**Comment by isontheline** [comment]
@macjl Do you have some time to beta test the latest release? 
https://testflight.apple.com/join/QSrBK59z

**Comment by macjl** [comment]
Bonjour ;)

It doesn't look like an IPv4 !

![photo_2024-09-30_14-00-31](https://github.com/user-attachments/assets/17ebaf1a-4700-409e-9f48-450d168878be)

**Comment by isontheline** [comment]
It does! :) 


It’s NAT64, your 5G connection does not provide you an IPv4 link : https://en.wikipedia.org/wiki/NAT64


You can use a dual IPv6/IPv4 link to ping a true real IPv4.

**Comment by macjl** [comment]
You’re right ! 
Tested it at home, and everything works as expected.
Thank you.
